### PR TITLE
feat: oci_pull supports loading Helm charts

### PR DESCRIPTION
**Pull request by sobotklp**
This MR shows an example of loading a Helm chart using `oci_pull` from a repository (for example, ECR)

For example, in `WORKSPACE`:
```
oci_pull(
    name = "dependency-helm-chart",
    digest = "sha256:865647e7904a1e4b4e130b423d20b34dcc89a5a37b7bc82e30ff100c30cfb86f",
    registry = "111222333444.dkr.ecr.us-east-2.amazonaws.com",
    repository = "helm/dependency-charts",
)
```

This is related to issue #282

### CODE REVIEW

File: oci/private/pull.bzl [hunk 1]
```diff
@@ -358,7 +359,12 @@ def _oci_pull_impl(rctx):
 
     mf, mf_len = downloader.download_manifest(rctx.attr.identifier, "manifest.json")
 
-    if mf["mediaType"] in _SUPPORTED_MEDIA_TYPES["manifest"]:
+    if "mediaType" in mf:
+        media_type = mf["mediaType"]
+    else:
+        media_type = mf["config"]["mediaType"]
```
Comment (alexeagle) [inline oci/private/pull.bzl:365]: it feels like this now needs a final else clause when the media_type couldn't be found in either of these two paths within the data structure

File: oci/private/pull.bzl [hunk 2]
```diff
@@ -391,6 +397,11 @@ def _oci_pull_impl(rctx):
         if hash not in tars:
             tars.append(hash)
 
+    if "mediaType" in image_mf:
```
Comment (alexeagle) [inline oci/private/pull.bzl:400]: could you extract a function here so we only need to have this logic once?

File: oci/private/pull.bzl [hunk 3]
```diff
@@ -183,6 +183,7 @@ _SUPPORTED_MEDIA_TYPES = {
     "manifest": [
         "application/vnd.docker.distribution.manifest.v2+json",
         "application/vnd.oci.image.manifest.v1+json",
+        "application/vnd.cncf.helm.config.v1+json",
```
Comment (alexeagle) [inline oci/private/pull.bzl:186]: can you come up with an example we could add to /WORKSPACE so there's a local reproduction we can use to test? I tried the literal helm chart you put in the PR description but get a 401 fetching that, I assume it's in a private registry.

**Review by alexeagle — COMMENTED** [review]
Thanks for the PR!!

Could you also try updating a documentation spot on oci_pull so it's easier for users to discover this ability?

**Comment by sobotklp** [comment]
@alexeagle no worries. I will spend some time adding more documentation and test cases as well. 

I'm also having trouble coming up with a general use case for someone (besides me) who wants to use this functionality. Downloading the OCI helm chart is easy enough, but what to do with it afterwards? Any guidance you could give here would be appreciated.

**Comment by thesayyn** [comment]
That's the rest of the iceberg, aka OCI artifacts, which we are not yet ready to support. Neither rules_oci nor crane is able to do anything with artifacts yet. 

On top of my head, I can think of using parts of a pulled helm chart to create a new one and push it with `oci_push`. 

Currently, rules_oci only conforms to the non-artifact part of https://github.com/opencontainers/image-spec, but we are considering adding full support for artifacts in the future.

**Comment by anguslees** [comment]
> Downloading the OCI helm chart is easy enough, but what to do with it afterwards?

I want this, fwiw!  I have my own bazel code that can do `helm template $chart.tgz`, but I would like to use rules_oci to actually fetch the chart file from an OCI repo.

**Comment by thesayyn** [comment]
I don't think we can accept this now, as the whole story for OCI artifacts is missing here. We need to work on https://github.com/bazel-contrib/rules_oci/issues/282 first before we can add this.

**Pull request closed without merging**